### PR TITLE
Add alpha, beta, and rc release definitions

### PR DESCRIPTION
**Pull request by colin-axner**
## Description

The proposed definitions for each phase of our release cycle. Please feel free to adjust my wording

closes: #881 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -21,6 +21,38 @@ To summarize: **All our ibc-go releases allow chains to communicate successfully
...
 
+## Release cycle
+
+IBC-Go follows a traditional release cycle involving an alpha, beta, and rc (release candidate) releases before finalizing a new version. As ibc-go works in a non-traditional area, we apply our own interpretation to each release type. We reserve the right to make both go API breaking changes and state machine breaking changes throughout the entire release cycle. The stable release guarentees do not go into affect until a final release is performed. 
+
+It is never advisable to use a non-final release in production. 
+
+### Alpha
+
+Alpha releases are intended to make avaliable new features as soon as they are functional. No correctness guarentees are made and alpha releases MAY contain serious security vulnerabilities, bugs, and lack of user tooling. Alpha releases may be made with known bugs and security vulnerabilities (so long as they don't affect the core functionality).
```
Comment (crodriguezvega) [inline RELEASES.md:32]:
```suggestion
Alpha releases are intended to make available new features as soon as they are functional. No correctness guarentees are made and alpha releases **may** contain serious security vulnerabilities, bugs, and lack of user tooling. Alpha releases may be made with known bugs and security vulnerabilities (so long as they don't affect the core functionality).
```

Comment (crodriguezvega) [inline RELEASES.md:32]: I don't get what this means `Alpha releases may be made with known bugs and security vulnerabilities (so long as they don't affect the core functionality).` considering that the previous sentence already said `No correctness guarentees are made and alpha releases MAY contain serious security vulnerabilities, bugs, and lack of user tooling`

**Review by crodriguezvega — APPROVED** [review]
Nice!

File: RELEASES.md [hunk 2]
```diff
@@ -21,6 +21,38 @@ To summarize: **All our ibc-go releases allow chains to communicate successfully
...
+
+Initial users of alpha releases are expected to be advanced, patient, and capable of handling unusal errors. Very basic integration testing will be performed by the ibc-go development team before alpha releases.
```
Comment (damiannolan) [inline RELEASES.md:34]:
```suggestion
Initial users of alpha releases are expected to be advanced, patient, and capable of handling unusual errors. Very basic integration testing will be performed by the ibc-go development team before alpha releases.  
```

File: RELEASES.md [hunk 3]
```diff
@@ -21,6 +21,38 @@ To summarize: **All our ibc-go releases allow chains to communicate successfully
...
+
+When the development team has determined a release is ready to enter the RC phase, a final security audit should be performed. The security audit should be limited to looking for bugs and security vulnerabilities. Code improvements may be noted, but they should not be acted upon unless highly desirable. 
+
+### RC 
+
+RC's are release candidates. Final releases should contain little to no changes in comparison to the latest RC. Changes included in between RC releases should be limited to:
+- improved testing
+- UX additions
+- bug fixes
+- highly requested changes by the community
```
Comment (damiannolan) [inline RELEASES.md:52]: Minor nit: Should we maybe capitalise the start of each bullet. E.g. "Improved testing", "Bug fixes"

**Review by damiannolan — APPROVED** [review]
Great work!

File: RELEASES.md (hunk 1 above)
Comment (colin-axner) [reply RELEASES.md]: updated

**Review by seantking — APPROVED** [review]
Great!

**Pull request merged**